### PR TITLE
diag(gsd): log basePath/mDir/existsSync when ready-phrase nudge fires

### LINKED ISSUE #4648: Stale dirListCache causes ready-phrase rejection despite files existing on disk
## Problem

`maybeHandleReadyPhraseWithoutFiles` in `guided-flow.js` rejects the LLM's "Milestone ready" signal because it reads the filesystem through a stale in-memory directory cache (`dirListCache` in `paths.js`). The LLM writes `M###-CONTEXT.md` and `M###-ROADMAP.md` via tool calls during the same turn, but `resolveMilestoneFile` -> `resolveFile` -> `cachedReaddir` returns the old (empty) directory listing. After 3 rejections the system gives up, even though the files exist on disk.

## Root Cause

**`guided-flow.js:286-301`** -- `maybeHandleReadyPhraseWithoutFiles` calls `resolveMilestoneFile()` without first invalidating the path cache.

Call chain:
1. `maybeHandleReadyPhraseWithoutFiles` -> `resolveMilestoneFile(basePath, milestoneId, "CONTEXT")` (line ~298)
2. -> `paths.js:418` `resolveMilestoneFile` -> `resolveFile(mDir, milestoneId, suffix)` (line ~422)
3. -> `paths.js:185` `resolveFile` -> `cachedReaddir(dir)` (line ~190)
4. -> `paths.js:96` `cachedReaddir` returns from `dirListCache` if the directory was listed before

`clearPathCache()` exists (`paths.js:125`) and is called in auto-dispatch, workflow-reconcile, and state transitions -- but **never** in `handleAgentEnd` or either of the two file-checking functions it calls.

The same bug affects `checkAutoStartAfterDiscuss` (`guided-flow.js:146-148`).

```javascript
// guided-flow.js:286 -- current (buggy)
export function maybeHandleReadyPhraseWithoutFiles(event) {
    // ...
    // BUG: reads through stale dirListCache -- no clearPathCache() call
    const contextFile = resolveMilestoneFile(basePath, milestoneId, "CONTEXT");
    const roadmapFile = resolveMilestoneFile(basePath, milestoneId, "ROADMAP");
    if (contextFile || roadmapFile) return false;  // never reached due to stale cache
    // ... proceeds to reject the ready signal
}
```

## Expected Behavior

Add `clearPathCache()` before the file existence checks in both `maybeHandleReadyPhraseWithoutFiles` (~line 297) and `checkAutoStartAfterDiscuss` (~line 145):

```javascript
import { clearPathCache } from "./paths.js";

// Before the artifact check:
clearPathCache();  // LLM may have written files this turn
const contextFile = resolveMilestoneFile(basePath, milestoneId, "CONTEXT");
const roadmapFile = resolveMilestoneFile(basePath, milestoneId, "ROADMAP");
```

This ensures fresh filesystem reads after the LLM's tool calls have written the milestone artifacts.

## Environment
- GSD version: 2.77.0
- Model: claude-opus-4-6
- Unit: discuss-milestone / M017

## Reproduction Context

After completing M016 (16 milestones, 55 slices, 43 tasks), GSD transitioned to M017. During the discuss phase the LLM wrote `M017-CONTEXT.md` and called `gsd_plan_milestone` (which writes `M017-ROADMAP.md`). On signaling "Milestone M017 ready.", the system rejected it 3 times claiming files were missing. Files confirmed present on disk via `ls`.

## Forensic Evidence

- LLM signaled ready 3 times; each time `maybeHandleReadyPhraseWithoutFiles` returned `true` (files missing) despite files existing on disk
- The `dirListCache` in `paths.js` was populated when the milestone directory was first listed (empty), and never invalidated before the file checks in `guided-flow.js`
- `clearPathCache()` is called in other code paths (auto-dispatch, state transitions) but not in the agent-end handler
- Historical stuck-loop anomalies for M011/S01 (2x), M013/S01 (3x), M014/S01 (3x) may share the same root cause -- stale cache causing artifact verification to fail

---
*Auto-generated by `/gsd forensics`*

### LINKED ISSUE #5197: [Bug]: Regression of #4648 — ready-phrase rejection still fires after `clearPathCache()` flush
## Summary

`maybeHandleReadyPhraseWithoutFiles` (`guided-flow.ts:581`) still falsely reports `M###-CONTEXT.md` and `M###-ROADMAP.md` as missing — pushing the LLM into a defensive re-call of `gsd_plan_milestone` whose UPSERT can overwrite the prior plan. This is the same symptom #4648 was filed for, after the fix that closed #4648 (the `clearPathCache()` call at `bootstrap/agent-end-recovery.ts:106`).

## Bug type

Regression (worked before, now fails)

## Steps to reproduce

Observed in `~/Github/test-apps/test123` during a `discuss-milestone` → `plan-milestone` auto handoff for M001. No deterministic minimal repro yet — this issue is filed to land diagnostic logging that will produce one.

1. `/gsd new-project --deep` for a small HTML/CSS/JS app
2. Auto-mode runs through deep setup → discuss-milestone → plan-milestone
3. LLM emits "Milestone M001 ready." after writing CONTEXT.md and calling `gsd_plan_milestone`
4. Recovery rejects the ready signal claiming files are missing
5. LLM is nudged to re-call `gsd_plan_milestone` — defensive re-call risks UPSERT-overwriting the prior plan

## Expected behavior

After `gsd_plan_milestone` writes ROADMAP.md and the LLM emits the ready phrase, recovery should observe both files on disk and skip the nudge.

## Actual behavior

Recovery reports both files missing and fires the nudge even when both files exist on disk.

## GSD version

2.78.1-dev.b0759e59b (also reported on 2.77.0 in #4648)

## Node.js version

v25.9.0

## Operating system

macOS 15.4 (Darwin 25.2.0)

## Install method

npm link / local checkout

## Affected area

Auto-mode / dispatch loop

## Model and provider

anthropic/claude-sonnet-4-6 (also seen on claude-opus-4-6 in #4648)

## Logs, screenshots, and evidence

Forensic state of `~/Github/test-apps/test123/.gsd/` after the failure:

```
# Files exist on disk:
-rw-r--r--  1 staff  5112 Apr 30 22:25 milestones/M001/M001-CONTEXT.md
-rw-r--r--  1 staff  1506 Apr 30 22:25 milestones/M001/M001-ROADMAP.md

# DB row populated:
sqlite> SELECT id, title, status, vision, requirement_coverage FROM milestones;
M001|Core Todo App|active|<vision>|Covers: R001, R002, R003, R004, R005. Out of scope by design: R006, R007, R008.

# Slices populated:
sqlite> SELECT id, milestone_id, title, status, sequence FROM slices;
S01|M001|Core Functionality|pending|1
S02|M001|Polish and Cleanup|pending|2

# Event log shows plan_milestone ran exactly once:
event-log.jsonl:
  {"v":2,"cmd":"plan-milestone","params":{"milestoneId":"M001"},"ts":"2026-05-01T03:25:31.492Z","actor":"agent",...}

# Recovery rejected 26 seconds AFTER plan_milestone committed:
notifications.jsonl:
  {"ts":"2026-05-01T03:25:57.727Z","severity":"warning","message":"Milestone M001: \"ready\" signal rejected — .gsd/milestones/M001/M001-CONTEXT.md and .gsd/milestones/M001/M001-ROADMAP.md are missing. Asking the LLM to complete the writes."}
```

The `clearPathCache()` flush at `agent-end-recovery.ts:106` (the fix that closed #4648) runs before this guard. `nativeTreeCache` is dead code in this branch (`getNativeTree()` at `paths.ts:32` is never called). So the false-missing has a different cause than #4648.

## Impact

- Severity: triggers redundant `gsd_plan_milestone` calls that risk UPSERT-overwriting the prior plan
- Frequency: at least one repro in test123; likely contributing to historical stuck-loop anomalies
- Consequence: data corruption risk if the LLM fabricates plan content to satisfy the tool's required fields, plus wasted tokens

## Additional information

Plausible causes worth investigating once diagnostics land (per Codex peer review):

1. **basePath/path-identity mismatch** — `_getPendingAutoStart()` is keyed on basePath; if the entry was registered with a different path (realpath vs raw, worktree vs project, symlink), `resolveMilestonePath` walks the wrong directory
2. **Module instance split** — `guided-flow.ts` loaded by two require paths means `pendingAutoStartMap` and the path cache live on different instances
3. **`checkAutoStartAfterDiscuss()` returning false for an unrelated reason** — recovery proceeds regardless
4. **Symlink/realpath mismatch** between where the LLM wrote and where recovery looks

The diagnostic PR (link below) logs `basePath`, `mDir`, and uncached `existsSync` results on the canonical paths whenever the nudge fires. When the false-positive recurs, the warning will tell us in one log line whether `existsSync` sees the files (→ resolver bug) or doesn't (→ basePath/wrong-directory bug).

The PR also adds a soft-brake to `prompts/plan-milestone.md` that conditions the re-call on ROADMAP.md *plus* DB slice rows — defense-in-depth against UPSERT overwrite of the partial-write case.

Related: #4648 (closed; original report and `clearPathCache()` fix)

### PR DESCRIPTION
**Pull request by jeremymcs**
## Linked issue

Closes #5197

- [x] I have linked an issue above. I understand that PRs without a linked issue will be closed without review.

---

## TL;DR

**What:** Adds diagnostic logging on the ready-phrase reject path in `maybeHandleReadyPhraseWithoutFiles` and a prompt-level soft-brake in `plan-milestone.md`.
**Why:** The fix that closed #4648 (the `clearPathCache()` flush at `agent-end-recovery.ts:106`) does not cover the failure mode observed in test123 — recovery still rejects ready signals while both artifacts exist on disk. See #5197.
**How:** Log `basePath`, `mDir`, and uncached `existsSync` results for both canonical paths when the nudge fires; add a prompt guard so the LLM does not defensively re-call `gsd_plan_milestone` when ROADMAP.md and DB slice rows already exist.

## What

Three files changed:

- `src/resources/extensions/gsd/guided-flow.ts` — adds a diagnostic `logWarning` block in `maybeHandleReadyPhraseWithoutFiles` immediately before the readyRejectCount increment. Logs `basePath`, the resolved `mDir`, the canonical CONTEXT/ROADMAP paths, and the result of `existsSync` on each (uncached). Wrapped in try/catch so the diagnostic never breaks the recovery path.
- `src/resources/extensions/gsd/prompts/plan-milestone.md` — adds an "Already Planned? Soft Brake" section right after `{{inlinedContext}}`. If `{{outputPath}}` already exists with at least one slice line AND `gsd_query` returns slice rows, skip the re-call and emit the ready phrase.
- `src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts` — two regression tests that assert the diagnostic warning is logged with the expected fields when the nudge fires (basePath, mDir, ctx-exists, roadmap-exists, canonical paths under the milestone directory).

## Why

#5197 documents the failure mode in detail. Short version: in test123 the recovery at `guided-flow.ts:581` reported both `M001-CONTEXT.md` and `M001-ROADMAP.md` missing 26 seconds after `gsd_plan_milestone` committed both files plus the DB rows. The `clearPathCache()` flush that closed #4648 ran first. `nativeTreeCache` is dead code in this branch. So the false-missing has a different cause than #4648 — most likely a basePath/path-identity mismatch in the pending auto-start entry, a module-instance split on the cache, or a symlink/realpath divergence (per Codex peer review).

The downstream consequence is real: when the LLM is nudged to re-call `gsd_plan_milestone`, and it doesn't have the original decomposition reasoning in context, it can fabricate stub content that overwrites the prior plan via the tool's UPSERT semantics.

## How

Rather than ship a speculative fix on a wrong theory, this PR lands instrumentation that will produce a definitive answer next time the bug fires. The diagnostic captures both:

- `mDir` from `resolveMilestonePath` — to detect whether path resolution itself is missing the directory
- `existsSync` on the canonical filename paths — uncached probe of the actual disk state, regardless of any cache state

When the false-positive recurs, one log line will distinguish:
- `ctx-exists=true, roadmap-exists=true` but the resolver returned null → resolver/cache bug
- `ctx-exists=false, roadmap-exists=false` but files visible to `ls` → basePath/wrong-directory bug
- `mDir=null` → milestone directory itself unreachable from the entry's `basePath`

The prompt soft-brake is defense-in-depth: even if the recovery nudges the LLM unnecessarily, the LLM will refuse the redundant re-call when ROADMAP.md and DB slice rows are both present. Conditioned on both per peer-review feedback so it doesn't fire in the legitimate partial-write case.

## Change type

- [x] `fix` — Bug fix (diagnostic + prompt guard)
- [x] `test` — Adding or updating tests

## Scope

- [x] `gsd extension` — GSD workflow

## Breaking changes

- [x] No breaking changes

## Test plan

- [x] CI passes
- [x] New/updated tests included
- [x] Manual testing — verified on the test123 forensic state described in #5197

Test results locally:

```
✔ 76 passed, 0 failed, 0 skipped
```

(ready-phrase-no-files-4573, headless-milestone-parity, plan-milestone-title, prompt-contracts test files combined.)

## AI disclosure

- [x] This PR includes AI-assisted code. Diagnosis, logging change, prompt edit, and tests authored with Claude Opus 4.7 (1M context). Diagnosis was peer-reviewed via the Codex CLI before patching — Codex caught a critical error in the original cache theory (the fix at `agent-end-recovery.ts:106` already addresses the cache angle), which redirected this PR from a speculative resolver fix to a logging-first approach.

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced diagnostic warnings for artifact recovery scenarios with improved verification
  * Refined question detection to better identify conditional user queries with varied punctuation
  * Added safeguard to prevent redundant milestone re-planning when plans already exist

* **Tests**
  * Extended test coverage for artifact recovery diagnostics and question detection edge cases

### CODE REVIEW

**Comment by jeremymcs** [comment]
⏳ **Accepted** — this comment requires code changes. Queuing fix...

File: src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts [hunk 1]
```diff
@@ -219,6 +220,81 @@ describe("#4573 maybeHandleReadyPhraseWithoutFiles", () => {
...
+      assert.equal(handled, true);
+
+      const logs = drainLogs();
+      const diag = logs.find(
+        (e) => e.component === "guided" && /ready-phrase-reject diagnostic/.test(e.message),
+      );
+      assert.ok(diag, "expected diagnostic warning to be logged when nudge fires");
+      assert.match(diag!.message, /mid=M001/);
+      assert.match(diag!.message, new RegExp(`basePath=${base.replace(/[/\\]/g, "[/\\\\]")}`));
+      assert.match(diag!.message, /mDir=/);
```
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

**Comment by jeremymcs** [comment]
⏳ **Accepted** — this comment requires code changes. Queuing fix...

File: src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...

[the remark this answers is not in the document]
Comment (jeremymcs) [reply src/resources/extensions/gsd/tests/ready-phrase-no-files-4573.test.ts:250]: ⏳ **Accepted** — this comment requires code changes. Queuing fix...
